npm/webpack-dev-server/src/makeWebpackConfig.ts

```diff
@@ -94,6 +92,42 @@ export function makeWebpackConfig (
     devServerEvents,
   } = config.devServerConfig
 
+  let configFile: string | undefined = undefined
+
+  if (!userWebpackConfig && !frameworkWebpackConfig) {
+    debug('Not user or framework webpack config received. Trying to automatically source it')
+
+    const { default: findUp } = await importModule('find-up')
+
+    configFile = await findUp(configFiles, { cwd: projectRoot } as { cwd: string })
+
+    if (configFile) {
+      debug('found webpack config %s', configFile)
+      const sourcedConfig = configFile.endsWith('mjs') ? await import(configFile) : require(configFile)
+
+      debug('config contains %o', userWebpackConfig)
+      if (sourcedConfig && typeof sourcedConfig === 'object') {
+        userWebpackConfig = sourcedConfig.default ?? sourcedConfig
+      }
```

Comment:
It's common for configs to be functions, can we refactor to something like this:

```suggestion
      if (sourcedConfig) {
        const source = sourcedConfig.default ?? sourcedConfig

        if (typeof source === 'object') {
          userWebpackConfig = source
        }

        if (typeof source === 'function') {
          userWebpackConfig = source()
        }
      }
```

Reply:
Is there a standard for what should be passed if the config is a function? I figured that could be a separate follow-up PR to this since it might need more definition around the behavior, and in the meantime you could just import the function it into your config file and invoke it as you'd expect.

Reply:
I'm cool with it being a followup, here is the docs for the function signature:
https://webpack.js.org/configuration/configuration-types/#exporting-a-function